x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/trainedmodel/ensemble/LogisticRegression.java

```diff
@@ -78,31 +78,29 @@ public Integer expectedValueSize() {
     }
 
     @Override
-    public List<Double> processValues(List<Double> values) {
+    public double[] processValues(double[][] values) {
         Objects.requireNonNull(values, "values must not be null");
-        if (weights != null && values.size() != weights.length) {
+        assert values[0].length == 1;
+        if (weights != null && values.length != weights.length) {
             throw new IllegalArgumentException("values must be the same length as weights.");
         }
         double summation = weights == null ?
-            values.stream().mapToDouble(Double::valueOf).sum() :
-            IntStream.range(0, weights.length).mapToDouble(i -> values.get(i) * weights[i]).sum();
+            Arrays.stream(values).mapToDouble(vs -> vs[0]).sum() :
+            IntStream.range(0, weights.length).mapToDouble(i -> values[i][0] * weights[i]).sum();
         double probOfClassOne = sigmoid(summation);
         assert 0.0 <= probOfClassOne && probOfClassOne <= 1.0;
-        return Arrays.asList(1.0 - probOfClassOne, probOfClassOne);
+        return new double[] {1.0 - probOfClassOne, probOfClassOne};
```

Comment:
This isn't quite right: we turn the set of values into a collection of predicted probabilities via the softmax function, i.e. the i'th predicted probability is `exp(values[i]) / sum_j{ exp(values[j]) }`. I think it is also worthwhile dividing through by `k = exp(max_j{ values[j] })` to handle the case the exp overflows: whence `exp(values[i] - k) / sum_j (exp(values[j] - k))`.

Reply:
@tveasey I will adjust to allow `logistic_regression` to be binomial and multinomial. 

Your concerns about overflow are handled in the already existing `Statistics#softMax` function. 

Thanks for the feedback!